autorest/azure/token.go

```diff
@@ -301,14 +301,17 @@ func (spt *ServicePrincipalToken) refreshInternal(resource string) error {
 	}
 
 	var newToken Token
+	var aadError AADServiceError
 	err = autorest.Respond(resp,
 		autorest.WithErrorUnlessOK(),
-		autorest.ByUnmarshallingJSON(&newToken),
+		autorest.ByUnmarshallingSuccessandErrorJSON(&newToken, &aadError),
```

Comment:
@ahmetalpbalkan is correct. I suggest you create a Responder, such as WithJAADServiceErrorUnlessOK, that unpacks the Body into an AADServiceError. Then use the normal ByUnmarshallingJSON to unpack the retrieved token.